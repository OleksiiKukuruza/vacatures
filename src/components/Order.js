import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import OrderValues from './OrderValues';
import Button from './Button';
import { Link } from 'react-router-dom';
import FeatureList from './FeatureList';

export const StyledOrder = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: ${props => props.theme.primary.main};
  color: ${props => props.theme.primary.contrastText};
  padding: ${props => 2 * props.theme.spacingUnit}px;
`;

const StyledExtraInfo = styled.div`
  display: flex;
  flex-direction: column;
`;

export const JobImg = styled.img`
  width: 100%;
  margin: ${props => 4 * props.theme.spacingUnit}px 0;
`;

export const StyledButton = styled(Button)`
  margin-top: ${props => 2 * props.theme.spacingUnit}px;
`;

const Order = ({
  vacancy: {
    id,
    name,
    features,
    originalPrice,
    discount,
    image
  }
}) => (
  <StyledOrder>
    <StyledExtraInfo>
      Bestelling
      <JobImg src={image} alt="Job Image" />
      {name}
      <FeatureList features={features} />
    </StyledExtraInfo>
    <div>
      <OrderValues
        name={name}
        originalPrice={originalPrice}
        discount={discount}
      />
      <Link to={`/${id}`}>
        <StyledButton fullWidth>
          Bestellen
        </StyledButton>
      </Link>
    </div>
  </StyledOrder>
);

Order.propTypes = {
  vacancy: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    originalPrice: PropTypes.number.isRequired,
    discount: PropTypes.number.isRequired,
    features: PropTypes.arrayOf(PropTypes.string).isRequired
  }).isRequired,
};

export default Order;
