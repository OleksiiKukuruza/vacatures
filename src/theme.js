const theme = {
  primary: {
    main: '#54b6c8',
    contrastText: '#fff'
  },
  background: '#f1f4f5',
  spacingUnit: 8
};

export default theme;
