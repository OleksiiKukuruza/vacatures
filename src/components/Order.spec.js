import React from 'react';
import { shallow } from 'enzyme';
import theme from '../theme';
import Order, { StyledOrder, StyledFeatures, StyledButton, JobImg } from './Order';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('Order', () => {
  it('renders Order without crashing', () => {
    const props = {
      vacancy: {
        id: '1',
        name: 'name',
        originalPrice: 100,
        discount: 10,
        image: 'img.src',
        features: ['First', 'Second']
      }
    };
    const { wrapper } = setup(Order, props);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledOrder without crashing', () => {
    const { wrapper } = setup(StyledOrder, {theme});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledFeatures without crashing', () => {
    const { wrapper } = setup(StyledFeatures, {theme});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledButton without crashing', () => {
    const { wrapper } = setup(StyledButton, {theme});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders JobImg without crashing', () => {
    const { wrapper } = setup(JobImg, {theme});
    expect(wrapper).toMatchSnapshot();
  });
});
