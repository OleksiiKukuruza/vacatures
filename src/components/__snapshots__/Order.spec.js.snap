// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order renders JobImg without crashing 1`] = `
.c0 {
  width: 100%;
  margin: 32px 0;
}

<img
  className="c0"
/>
`;

exports[`Order renders Order without crashing 1`] = `
<styled.div>
  <styled.div>
    Bestelling
    <styled.img
      alt="Job Image"
      src="img.src"
    />
    name
    <FeatureList
      features={
        Array [
          "First",
          "Second",
        ]
      }
    />
  </styled.div>
  <div>
    <OrderValues
      discount={10}
      name="name"
      originalPrice={100}
    />
    <Link
      replace={false}
      to="/1"
    >
      <Styled(styled.button)
        fullWidth={true}
      >
        Bestellen
      </Styled(styled.button)>
    </Link>
  </div>
</styled.div>
`;

exports[`Order renders StyledButton without crashing 1`] = `
.c0 {
  margin-top: 16px;
}

<styled.button
  className="c0"
  theme={
    Object {
      "background": "#f1f4f5",
      "error": "#f44336",
      "primary": Object {
        "contrastText": "#fff",
        "main": "#54b6c8",
      },
      "spacingUnit": 8,
    }
  }
/>
`;

exports[`Order renders StyledOrder without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #54b6c8;
  color: #fff;
  padding: 16px;
}

<div
  className="c0"
/>
`;
